src/edu/washington/escience/myria/operator/agg/MultiGroupByAggregate.java

```diff
@@ -43,6 +49,11 @@
   private List<TupleBatch> groupKeyList;
   /** Holds the corresponding aggregation state for each group key in {@link #groupKeys}. */
   private transient List<Object[]> aggStates;
+  /** Holds the corresponding TB for each group key in {@link #groupKeys}. */
+  private transient List<List<TupleBatch>> tbgroupState;
+  /** Holds the bitset for each group key in {@link #groupKeys}. */
+  HashMap<Integer, BitSet> bs = new HashMap<Integer, BitSet>();
```

Comment:
Just a heads-up that these keys will be replace by rows in primitive columns in my refactor_aggregate PR.